### PR TITLE
CEDS-2052 Back button navigation on change from summary page 

### PR DESCRIPTION
**Pull request by bambuchaAdm**
This is first PR. Works till goods location. It is enough big now and it still work to do.

### CODE REVIEW

**Comment by bambuchaAdm** [comment]
please rebuild

**Comment by bambuchaAdm** [comment]
retest this please

**Comment by bambuchaAdm** [comment]
rebuild this please

File: app/controllers/navigation/Navigator.scala [hunk 1]
```diff
@@ -41,6 +43,17 @@ class Navigator @Inject()(appConfig: AppConfig, auditService: AuditService) {
       case _ => Results.Redirect(call)
     }
 
+  def continueTo(mode: Mode, factory: Mode => Call)(implicit req: JourneyRequest[AnyContent], hc: HeaderCarrier): Result =
+    redirectTo(mode.next, factory)
```
Comment (PatrykRudnicki) [inline app/controllers/navigation/Navigator.scala]: Why we need `continueTo` if it's just a wrapper for `redirectTo`?

File: app/controllers/navigation/Navigator.scala [hunk 2]
```diff
@@ -182,21 +195,52 @@ object Navigator {
     case page                  => throw new IllegalArgumentException(s"Navigator back-link route not implemented for $page on occasional")
   }
 
+  val universal: PartialFunction[DeclarationPage, Mode => Call] = {
```
Comment (PatrykRudnicki) [inline app/controllers/navigation/Navigator.scala]: Can we use the better name for it? If I'm understanding this correctly what do you think about `common`?

File: app/controllers/navigation/Navigator.scala (hunk 1 above)
Comment (bambuchaAdm) [reply app/controllers/navigation/Navigator.scala]: `mode.next` - redirect preserve mode - continue change it

File: app/controllers/navigation/Navigator.scala (hunk 2 above)
Comment (bambuchaAdm) [reply app/controllers/navigation/Navigator.scala]: Would be fixed in part2

**Pull request merged**